_includes/markdown/JavaScript.md

```diff
@@ -155,6 +155,80 @@ On all new projects you should be using up to date JavaScript methodologies comb
+
+There are several measures you can take to circumvent this XSS vulnerability:
+
+#### Use ```textContent``` instead of ```innerHTML```
+
+When setting the human-readable content of a single element, using [```textContent```](https://developer.mozilla.org/en-US/docs/Web/API/Node/textContent) is safer than using ```innerHTML``` because it does not parse strings as HTML—meaning any malicious code passed to it will not be executed.
+
+```javascript
+const anotherElement = document.getElementById( 'anotherElement' );
+
+anotherElement.textContent = 'This will get displayed as human-readable text, even if it contains HTML-like <syntax>.';
+```
+
+#### Use the ```Element``` API instead of HTML-parsing methods
+
+When you need to create multiple DOM elements, use JavaScript's [```Element```](https://developer.mozilla.org/en-US/docs/Web/API/Element) API to create elements, set attributes and append them to the document. Creating your own elements and attributes will ensure that only those you explicitly define will make their way into the DOM. Refactoring our first example:
+
+```javascript
+const someElement = document.getElementById( 'someElement' );
+const someUrl = 'https://someurl.com/';
+const someContent = 'Some content';
+
+// Create the new elements.
+const container = document.createElement( 'div' );
+const anchor = document.createElement( 'a' );
+
+// Set attributes and content.
+container.setAttribute( 'class', 'container' );
+anchor.setAttribute( 'href', someUrl );
+anchor.textContent = someContent;
+
+// Create structure for the new elements.
+container.appendChild( anchor );
+
+// Append new elements to the DOM all at once.
+someElement.appendChild( container );
+```
+
+Note that appending new elements to the DOM is a relatively expensive operation, so generally you'll want to build the structure of new, nested elements _before_ adding them to the DOM, then append them to an existing element all at once by appending a single container element.
+
+#### Sanitize HTML strings before appending to the DOM
+
+In general, using the ```Element``` API is the preferred best practice to safely create and add DOM elements. However, you'll notice that the refactored example is much more verbose than the original example—what was just a few lines of executable code more than doubled after refactoring. This can become painful if you need to create many new elements. In those cases, the convenience of methods like ```innerHTML``` can be extremely tempting. 
+
+If you need to generate a large amount of HTML dynamically, consider adding a client-side sanitization library to your project so you can strip potentially malicious code from your HTML before you add it to the DOM. Two popular libraries are [DOMPurify](https://github.com/cure53/DOMPurify) and [js-xss](https://github.com/leizongmin/js-xss). They work similarly:
+
+```javascript
```

Comment:
@dkoo If we want to sanction this library, that's fine with me, but I don't really want to have to maintain third party code blocks in our best practice documentation. Let's remove this block in favor of a link to documentation.